cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -121,22 +135,30 @@ public PayloadTable createPayloadTable(NamespaceId namespace, String tableName)
   }
 
   /**
-   * Creates a new instance of {@link HTable} for the given {@link TableId}. If the hbase table doesn't exist,
-   * a new one will be created with the given number of splits.
+   * Creates a new instance of {@link HTable} for the given {@link TableId}. If the hbase table doesn't
+   * exist, a new one will be created with the given number of splits.
    */
-  private HTable createTableIfNotExists(TableId tableId, int splits) throws IOException {
+  private HTable createTableIfNotExists(TableId tableId, int splits,
+                                        Class<? extends Coprocessor> coprocessor) throws IOException {
     // Create the table if the table doesn't exist
     try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
       // Set the key distributor size the same as the initial number of splits, essentially one bucket per split.
       AbstractRowKeyDistributor keyDistributor = new RowKeyDistributorByHashPrefix(new OneByteSimpleHash(splits));
 
-      HTableDescriptor htd = tableUtil
+      TableId metadataTableId = tableUtil.createHTableId(NamespaceId.SYSTEM, cConf.get(
```

Comment:
Also, I don't think we need to have it configurable as long as it is created in the cdap system namespace.